src/Kros.AspNetCore/Authorization/ClientCredentialsAuthorizationMiddleware.cs

```diff
@@ -0,0 +1,109 @@
+﻿using Kros.AspNetCore.Options;
+using Kros.Utils;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Primitives;
+using Microsoft.IdentityModel.Protocols;
+using Microsoft.IdentityModel.Protocols.OpenIdConnect;
+using Microsoft.IdentityModel.Tokens;
+using Microsoft.Net.Http.Headers;
+using System;
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Kros.AspNetCore.Authorization
+{
+    /// <summary>
+    /// Middleware for identity server authorization.
+    /// </summary>
+    internal class ClientCredentialsAuthorizationMiddleware
+    {
+        private readonly RequestDelegate _next;
+        private readonly ClientCredentialsAuthorizationOptions _authorizationOptions;
+        /// <summary>
+        /// Ctor.
+        /// </summary>
+        /// <param name="next">Next middleware.</param>
+        /// <param name="options">Client credentials authorization options.</param>
+        public ClientCredentialsAuthorizationMiddleware(RequestDelegate next, ClientCredentialsAuthorizationOptions options)
+        {
+            _next = Check.NotNull(next, nameof(next));
+            _authorizationOptions = Check.NotNull(options, nameof(options));
+            Check.NotNullOrWhiteSpace(options.AuthorityUrl, nameof(options.AuthorityUrl));
+            Check.NotNullOrWhiteSpace(options.Scope, nameof(options.AuthorityUrl));
+            Check.NotNull(options.ClockSkew, nameof(options.ClockSkew));
+        }
+
+        /// <summary>
+        /// HttpContext pipeline processing.
+        /// </summary>
+        /// <param name="httpContext">Http context.</param>
+        public async Task Invoke(HttpContext httpContext)
+        {
+            await ValidateJwtToken(httpContext);
+            await _next(httpContext);
+        }
+
+        private async Task ValidateJwtToken(HttpContext httpContext)
+        {
+            if (httpContext.Request.Headers.TryGetValue(HeaderNames.Authorization, out StringValues value))
```

Comment:
Toto parsovanie tokenu už máme urobené v JwtBearerClaimsMiddleware.cs takto:
`private static bool TryGetTokenValue(IHeaderDictionary headers, out string tokenValue)
        {
            if (headers.TryGetValue(HeaderNames.Authorization, out StringValues authHeader))
            {
                string authHeaderToken = authHeader.First();
                if (authHeaderToken.StartsWith(AuthTokenPrefix))
                {
                    tokenValue = authHeaderToken.Substring(AuthTokenPrefix.Length);
                    return true;
                }
            }
            tokenValue = null;
            return false;
        }`

Bolo by dobré použiť to, zváž vyseparovanie do Helpera JwtAuthorizationHelper